pkg/flag/report_flags.go

```diff
@@ -227,6 +237,12 @@ func (f *ReportFlagGroup) ToOptions() (ReportOptions, error) {
 		}
 	}
 
+	// "--so-summary" option is available only with "--format table".
+	if noSummaryTable && format != types.FormatTable {
+		noSummaryTable = false
+		log.Warn(`"--no-summary-table" can be used only with "--format table".`)
```

Comment:
For example, suppose a flag has already been added for a long time, and we don't want to terminate a program for backward compatibility as it will change behavior. In that case, we have to make it a warning, but an error is basically better because the user may not see the warning message.

```suggestion
		return xerrors.New(`"--no-summary-table" can be used only with "--format table".`)
```

Reply:
changed in https://github.com/aquasecurity/trivy/pull/8177/commits/786e704264f691391a0130536ce4256adacd583e